### ISSUE TITLE
Upgrade this project to Gradle 7.4.2, Java 17, Spock 2

### ISSUE THREAD

**Issue by martinda** [tudortimi/gradle-hdvl#61]
Upgrade this project to the current latest LTS. If there are no objections, I can take a stab at it.

For Spock 2, some cosmetic changes to the tests need to be done (temporary files), but they should be trivial.

**Comment by tudortimi** [comment]
The only thing I'm not 100% sure of is Java 17. I'm not sure what version Gradle bundles, but I would rely on that one. The idea is that this plugin is supposed to run on pretty barebones RHEL7 or RHEL8 machines, with no extra packages.

**Comment by tudortimi** [comment]
W.r.t. Gradle 7.4, I haven't been very strict about not using private APIs (with the intention of getting something out the door and fixing it later), so there may be some pain points.

**Comment by tudortimi** [comment]
I'm a big fan of small, focused PRs. This issue can surely be split into a Gradle and a Spock part.

**Comment by martinda** [comment]
Correct. Gradle bundles Kotlin and Groovy, but no the JVM. I would say it's common to have LTS versions: 8, 11 and 17.
